packages/ai-core/src/browser/frontend-prompt-customization-service.ts

```diff
@@ -17,13 +17,19 @@
 import { DisposableCollection, URI, Event, Emitter } from '@theia/core';
 import { OpenerService } from '@theia/core/lib/browser';
 import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
-import { PromptCustomizationService, PromptTemplate } from '../common';
+import { PromptCustomizationService, PromptTemplate, CustomAgentDescription } from '../common';
 import { BinaryBuffer } from '@theia/core/lib/common/buffer';
 import { FileService } from '@theia/filesystem/lib/browser/file-service';
 import { FileChangesEvent } from '@theia/filesystem/lib/common/files';
 import { AICorePreferences, PREFERENCE_NAME_PROMPT_TEMPLATES } from './ai-core-preferences';
 import { AgentService } from '../common/agent-service';
 import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import { load } from 'js-yaml';
+
+const templateEntry = `-id: my_agent
+name: My Agent
+description: This is an example agent. Please adapt the properties to fit your needs.
+prompt: You are an example agent. Be nice and helpful to the user.`;
```

Comment:
This template is invalid and breaks the whole functionality. The resulting template looks like this:

```yaml
-id: my_agent
name: My Agent
description: This is an example agent. Please adapt the properties to fit your needs.
prompt: You are an example agent. Be nice and helpful to the user.
```

This is not an array but an object with 4 parameters, including `-id`.

---

As we are using `js-yaml` we should use its functionality to generate the yaml from a json object and not do any string manipulation on our own. See https://github.com/nodeca/js-yaml?tab=readme-ov-file#dump-object---options-